### ISSUE TITLE
[ FF7 ] Collection of widescreen 16:9 bugs

### ISSUE THREAD

**Issue by CosmosXIII** [julianxhokaxhiu/FFNx#474]
This is the collection of bugs related to the new widescreen 16:9 aspect ratio mode.

**Common**
- [x] 16:9 aspect ratio stretches to fill the screen when displayed on an ultrawide monitor

**Field mode**
- [ ]  Some fields show a back bar on the top/bottom of the screen
- [ ]  The chocobo black screen in the Chocobo Farm does not show
- [ ]  The wind overlay in Gaea's cliff does not scale to fill the screen
- [ ]  In fship_22, the vertical scrolling stops too early
- [ ] Aerith's Great Gospel Limit 4 shows the black background in the edges

**Battle mode**
- [ ] Compact Battle UI has black background on the bottom right and left
- [x] Limit break boxes extend beyond the intended window
- [ ] Pollensalta at Northern Cave - The full screen spell texture is not following true widescreen

**Worldmap mode**

**Other modes**
- [x] shop characters/text to appear cut off
- [ ]  Condor minigame is still 4:3

Before reporting new bugs/glitches, make sure you are running the latest canary version of FFNx. Also double check if the bug or glitch is not already in this list

**Comment by KuraudoXS** [comment]
7th 2.6.14
FFNx 1.13.0.8

Other Mod - Compatibility issue
1) Compact Battle UI has black background right and left.

Fields
1) As soon as you step in the Highwind the animation background is not wide, resulting in black background right and left.

**Comment by AnsgarTOdinson** [comment]
16:9 mode stretches to fill the screen when displayed on an ultrawide monitor.  Native 4:3 mode does not.  This applies to all modes.

**Comment by tangtang95** [comment]
@KuraudoXS found this field map 'junbin21' having some problem with top camera range. Need to fix it in future config.toml

**Comment by Sdub-12** [comment]
**PLEASE NOTE:** If you come here to report a crash while you were using 7th Heaven, please do on their relative support channels. NO SUPPORT will be provided, and your issue will be close immediately.

**Describe the bug**
16:9 Mode is causing shop characters/text to appear incorrectly/moved. The issue seems to be specifically to shop prices. Will attach screenshots. Using Canary build.

**To Reproduce**

1. Go to Gamedriver Settings
2. Enable 16:9 Widescreen
3. Begin Game
4. Enter Shop interface
5. See error

**Expected behavior**
I expected the prices to appear normally as is with 4:3

**Screenshots**

![Shops with WS](https://user-images.githubusercontent.com/69642674/185484928-0553aba9-0979-49e6-8394-73c9bfc39c6f.jpg)
![Shops Without  WS](https://user-images.githubusercontent.com/69642674/185484933-1ef898cd-4f8f-4ec3-891c-4fd217407342.jpg)
![ZeroModNotUsing](https://user-images.githubusercontent.com/69642674/185484935-b8fb28e6-29ed-4c77-8dc2-8139d362fb02.png)
![ZeroModUsing](https://user-images.githubusercontent.com/69642674/185484932-03f05b9a-4f83-4906-a3e5-dfc35811598d.png)

Settings I was using
![settings](https://user-images.githubusercontent.com/69642674/185485056-8658493e-941b-42dd-a5f8-a057c16b5fe7.png)


**GPU (please complete the following information):**

- NVIDIA GeForce GTX 1060 3GB
- GeForce Game Ready Driver Version 516.94

**Additional context**
Feel free to reach out to me anytime, I am Sdub in the Qhimm discord

**Comment by julianxhokaxhiu** [comment]
@Sdub-12 when putting reports please make sure you tell which is the FFNx version you used, not the date. Thank you.

**Comment by Sdub-12** [comment]
> @Sdub-12 when putting reports please make sure you tell which is the FFNx version you used, not the date. Thank you.

Updated, I was unable to locate the number at the time, it was not displaying on the screen for borderless windowed mode, Was displayed when using Fullscreen. Apologies, Ive only been using the program for less than a week

**Comment by tangtang95** [comment]
@CosmosXIII Regarding the menu shop scissor glitch, it's probably not going inside this if condition https://github.com/julianxhokaxhiu/FFNx/blob/master/src/renderer.cpp#L1237. You did it also for swirl, but is it really necessary?

**Comment by CosmosXIII** [comment]
> @CosmosXIII Regarding the menu shop scissor glitch, it's probably not going inside this if condition https://github.com/julianxhokaxhiu/FFNx/blob/master/src/renderer.cpp#L1237. You did it also for swirl, but is it really necessary?

yeah, I fixed the conditions in this PR and the shop glitch is fixed now: https://github.com/julianxhokaxhiu/FFNx/pull/484
the swirl one did not need that condition so I removed it

**Comment by Sdub-12** [comment]
Canary
7th v2.6.1.20
FFNx v1.13.0.75 

Describe the bug
When doing anything with the Chocobo Farm (Moving, Breeding, Riding) It does not show the Black screen with the chocobo. It instead just shows the top of the barn with the information from the Chocobo as seen in the screenshots below.

To Reproduce
Have widescreen mode on
Attempting to use chocobo farm.

Expected behavior
To be able to see the traditional chocobo screen

Screenshots
![image](https://user-images.githubusercontent.com/69642674/188949600-b6e2fee5-bbbd-4d57-9054-c15d0398425d.png)
![image](https://user-images.githubusercontent.com/69642674/188949641-0426c845-aba8-49e6-849a-3c7b11bb7ec8.png)
![image](https://user-images.githubusercontent.com/69642674/188949661-85d3f990-9a7e-4744-a9bb-29c8ba8a0bed.png)
![image](https://user-images.githubusercontent.com/69642674/188949714-37479cb3-507d-428a-862e-4a916f29844e.png)

GPU (please complete the following information):

    NVIDIA GeForce GTX 1060 3GB
    GeForce Game Ready Driver Version 516.94

Additional context
Let me know if you need any further information.

**Comment by Neffina** [comment]
![Fight](https://user-images.githubusercontent.com/71432816/189304855-9cd814cb-8a72-4a27-bbef-a3a59207e543.jpg)


I noticed that in the penultimate boss fight in 16:9 format, the boss is not really displayed. The pens on his head are too far away. It's not because of the model, if 16:9 is deactivated, everything is correct.

**Comment by SirDevil99** [comment]
![Desktop Screenshot 2022 09 10 - 07 54 17 34-min](https://user-images.githubusercontent.com/113224568/189467782-0b7fb1a5-abe1-4d74-8844-9f41ffd65805.png)

This affect doesn't scale properly in Gaea’s Cliff as it is only a box as shown in the image above.

**Comment by chrcoluk** [comment]
If ESUI limit box issue is fixed I appreciate that, I was going to fix inside ESUI itself, so thank you.

**Comment by LaZar00** [comment]
In fship_22, when Cloud reach the tornado, we can see Highwind with Rufus, Heidegger, Scarlet, Hojo... In the 16:9 version, the scroll does not pans to the bottom when Hojo speaks. I leave 4:3 version and 16:9 version to see differences.

https://user-images.githubusercontent.com/46090031/191134187-e1a7a7ce-e24c-45c3-ae19-f460b35bfc6f.mp4


https://user-images.githubusercontent.com/46090031/191134190-80950061-373b-4a21-9045-3c518023205f.mp4

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.13.0.83
7H 2.6.1.25

I wouldn't say this is a bug, maybe an enhancement.

In trnad_3 (704), I think that the width of the field is enough wide to use the 16:9 proportion. But you must walk for scroll/pan horizontally the field view... I'm not sure if this is normal or it is as is.

Uploading FF7 2022.09.20 - 01.21.30.03.mp4…

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.13.0.83
7H 2.6.1.25

Another widescreen issue appears the first time we get Highwind (without Cloud). Your character is Tifa and you must go down to the bridge. Then there is a sequence where Barret and Cait are present and there is a pan movement of the field to the right, but then later, the camera stays and Barret and Cait disappear magically xD.

https://user-images.githubusercontent.com/46090031/191496995-ced2bce4-61c5-48c9-887b-3b9d3a2e2a42.mp4

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.13.0.84
7H 2.6.1.27

Great Gospel Limit 4 of Aeris shows the black background in the edges of the box:

![imagen](https://user-images.githubusercontent.com/46090031/191985453-cf552482-608e-4edc-afce-aa95a6577565.png)

**Comment by CosmosXIII** [comment]
@LaZar00 trnad_3 and trnad_4 have scrolling but not enough to fill widescreen. These ones remain 4:3 for now but we are working on solutions for those.

**Comment by LaZar00** [comment]
I was not sure if it would fill all the 16:9 aspect. But it's ok. I only reported it because I've seen some field backgrounds showing all the image even it does not fill 16:9 aspect ratio.
In any case, you are doing a great job with widescreen!

**Comment by julianxhokaxhiu** [comment]
Closed by mistake. Re-opened.

**Comment by LaZar00** [comment]
> Using: Canary FFNx 1.13.0.84 7H 2.6.1.27
> 
> Great Gospel Limit 4 of Aeris shows the black background in the edges of the box:
> 
> ![imagen](https://user-images.githubusercontent.com/46090031/191985453-cf552482-608e-4edc-afce-aa95a6577565.png)

About this issue, the model/box of the sky it is in magic.lgp. The RSD file using it is SKY_A.RSD, that uses the model SKY_A.P and the texture SKY_1.TEX.

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.14.0.1
7H 2.7.0.8

When you go for the Huge materia of Mt. Corel, you have to do the minigame of the little train. In this case, you have to go wagon by wagon until the locomotive and stop the train.
In the video you can see how part of the mountain (top left part of the image) is rendered at certain point, leaving some part without rendering. This happens in other camera views of this sequence also.

https://user-images.githubusercontent.com/46090031/193342093-f1b5d56e-fa3f-4753-8a5b-71fa5fd25f75.mp4

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.14.0.1
7H 2.7.0.8

Ok, another magic trick, this time with Young Cloud, like the one of Barret/Cait Sith in Highwind and Tifa.

https://user-images.githubusercontent.com/46090031/196009240-fd7aedfa-7d23-46d4-b0db-7c843c44194b.mp4

**Comment by chrcoluk** [comment]
Something that can affect trying to do WS on vanilla battle layout is if box width is wider than 4:3 size the scroll boxes stop rendering, let me know if you want a clip.

**Comment by tangtang95** [comment]
> Something that can affect trying to do WS on vanilla battle layout is if box width is wider than 4:3 size the scroll boxes stop rendering, let me know if you want a clip.

Yeah, if you have an image, or even a way to test it, it would help a lot

**Comment by LaZar00** [comment]
Using:
Canary
FFNx 1.14.0.1
7H 2.7.0.8

Well, the last positional bug I found for Widescreen. I think that three are all possible, but who knows. I have not checked 100% of the game.

https://user-images.githubusercontent.com/46090031/197620117-82fa8384-0a27-49f6-8fb4-5bc370fd5209.mp4

**Comment by LaZar00** [comment]
Canary
FFNx 1.15.0.46
7H 3.0.0.1

Found this issue with the sunrays layer of the field. I assume it is widescreen:
![imagen](https://user-images.githubusercontent.com/46090031/224669940-7308c042-c31c-47ef-91d2-3eaa7c58ecfd.png)

**Comment by KuraudoXS** [comment]
Describe the bug
The Honeybee Inn keyhole scenes don't display properly when widescreen is enabled. The black texture with the transparent keyhole in the center does not extend to cover the widescreen areas.

To Reproduce
Steps to reproduce the behavior: Enable widescreen, view those scenes.

Expected behavior
Widescreen should be force disabled for these scenes so they display correctly.

FYI From ChtonVII

**Comment by julianxhokaxhiu** [comment]
See also #620

**Comment by ChthonVII** [comment]
Unless I'm mistaken, no one has yet mentioned that the ultima spell's screen-filling glow effect fails to cover the whole screen in 16::9. Looks about the same as Pollensalta's skill before it was fixed.
